python/plugins/processing/algs/qgis/OpenVectorFile.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    OpenVectorFile.py
+    ---------------------
+    Date                 : October 2017
+    Copyright            : (C) 2017 by Etienne Trimaille
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Etienne Trimaille'
+__date__ = 'October 2017'
+__copyright__ = '(C) 2017, Etienne Trimaille'
+
+# This will get replaced with a git SHA1 when you do a git archive323
+
+__revision__ = '$Format:%H$'
+
+from os.path import splitext
+from qgis.core import (
+    QgsVectorLayer,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterString,
+    QgsProcessingOutputVectorLayer,
+    QgsProcessingException)
+
+from processing.algs.qgis.QgisAlgorithm import QgisAlgorithm
+
+
+class OpenVectorFile(QgisAlgorithm):
+
+    """This algorithm can open a vector file on the file system."""
+
+    INPUT_FILE = 'INPUT_FILE'
+    EXTRA_PARAMETERS = 'EXTRA_PARAMETERS'
+    OUTPUT = 'OUTPUT'
+
+    def group(self):
+        return self.tr('File tools')
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterFile(
+            self.INPUT_FILE, self.tr('Input vector file')))
+
+        self.addParameter(QgsProcessingParameterString(
+            name=self.EXTRA_PARAMETERS, optional=True,
+            description=self.tr('Parameters after the file path')))
```

Comment:
This should be a bit more verbose and specify what kind of parameters are expected